### DISCUSSION TITLE
Time series: number of y-axis ticks on small panels

### DISCUSSION THREAD

**Discussion by drewhammond** [grafana/grafana#39307] [Dashboards, Panels & Visualizations]
I'm not sure if this is a bug or a feature request, but the new time series panels have really awkward y-axis intervals when the panels are small.

In this (dramatic) example, I have no idea where 40 is relative to the min and max of the data without hovering over the panel. The old graph is much easier to interpret quickly.

<img width="882" alt="Screen Shot 2021-09-16 at 7 39 51 PM" src="https://user-images.githubusercontent.com/2405390/133702429-befeb3ff-ca00-4643-9c5e-b21894db049e.png">

If this is intentional, is there a way to customize the number of y-axis ticks so it behaves more like the old graph panel?

Most of our dashboards use small panels and this feels like a step in the wrong direction.

Thanks!

**Comment by leeoniya** [comment]
for sure we need to fix this. it's been secretly bugging me for a while. the old panel might actually have too many ticks -- i think for small panels even just two at the limits (+ maybe one in the middle) would be both, sufficient and cleaner.

i'll work to improve this by 8.3. not sure if this will go into Grafana core or uPlot core. i have some initial work in this direction already that tries to combine the previous logic with adjustable max tick density: https://leeoniya.github.io/uPlot/demos/nice-scale.html

opened https://github.com/grafana/grafana/issues/39359

**Reply by DmitrySidorow** [reply]
maybe this is also related?
https://github.com/grafana/grafana/issues/38849#issuecomment-913566059

**Reply by leeoniya** [reply]
related, yes. there are more things to work out holistically tho: https://github.com/grafana/grafana/issues/35838

**Reply by drewhammond** [reply]
Thanks for opening the issue! We're looking forward to 8.3.

FWIW, my personal preference is for a minimum of 3 ticks. I've never thought the density of the old graph panel was too much to handle, but our dashboards are very dense in general so that might just go with the theme.

Bonus points if max tick density can be configured somewhere 🙂

**Comment by leeoniya** [comment]
this likely won't make it into 8.3.0, since it turned out to require some pretty significant changes in how the underlying plotting library works. i've made pretty good progress in a branch [1], but not yet happy with the resulting API and performance implications, so it will need more work in the coming weeks. still aiming to get this into an 8.3.x release.

[1] https://github.com/leeoniya/uPlot/tree/ticks-axis-incrs

**Reply by dredvard** [reply]
Any chance that  tick marks and labels could be independent?  For log plots its really nice having lots of grid lines - but the labels make it unreadable.

![Screenshot 2021-11-22 224302](https://user-images.githubusercontent.com/1581799/142968748-1954fd32-ee39-43c9-8024-af92fe9f4c8f.png)

?

**Reply by rolf-epking** [reply]
This issue still exists with grafana 9.3.6

https://github.com/grafana/grafana/issues/63840

**Comment by briend** [comment]
We're still seeing this in Grafana `9.4.7`.  Having just the `0 B/s` mark is not helpful.  If we resize the panels a bit taller then it is fine, but we'd rather keep the real estate.

![image](https://user-images.githubusercontent.com/6015639/235252302-1dc40995-cf35-4af2-a4a5-7958ecccbecb.png)

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Seems to have been fixed in https://github.com/grafana/grafana/issues/63840. Feel free to make an [issue](https://github.com/grafana/grafana/issues) if you cannot find a solution and would like to request the feature.
